python/ray/autoscaler/util.py

```diff
@@ -58,6 +58,14 @@ def validate_config(config: Dict[str, Any]) -> None:
     except jsonschema.ValidationError as e:
         raise jsonschema.ValidationError(message=e.message) from None
 
+    # Detect out of date defaults. This happens when the autoscaler that filled
+    # out the default values is older than the version of the autoscaler that
+    # is running on the cluster.
+    if "cluster_synced_files" not in config:
```

Comment:
Interesting, what would be the different between this and the schema validation above?

Reply:
the first validation has pretty much every field as optional based off what users actually have to provide. So it checks if the format of the user's cluster config is correct.

This second validation would be for internal use which marks almost everything as required based off the expectation that the user's config has been filled out with default values. This second check will check for any bugs where we are not filling out defaults correctly or version mismatch errors. It also generalizes my code a little so that contributors don't have to add a new if check for every time they add a new config.